ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -115,66 +137,93 @@ private static Object enforceFieldType(final JsonValueContext context) {
   }
 
   private static String processString(final JsonValueContext context) {
-    if (context.val instanceof Map) {
-      return context.deserializer.gson.toJson(context.val);
+    if (context.val instanceof ObjectNode) {
+      try {
+        // this ensure sorted order, there's an issue with Jackson where just enabling
+        // SORT_PROPERTIES_ALPHABETICALLY does not work if it is not a POJO-backed
+        // JSON object
+        return SORTED_MAPPER.writeValueAsString(
+            SORTED_MAPPER.treeToValue(context.val, Object.class)
+        );
+      } catch (JsonProcessingException e) {
+        throw new KsqlException("Unexpected inability to write value as string: " + context.val);
+      }
+    }
+    if (context.val instanceof ArrayNode) {
+      return Streams.stream(context.val.elements())
+          .map(val -> processString(
+              new JsonValueContext(context.deserializer, context.schema, val)
+          ))
+          .collect(Collectors.joining(", ", "[", "]"));
     }
-    return context.val.toString();
+    return context.val.asText();
   }
 
   private static Object enforceValidBytes(final JsonValueContext context) {
-    // before we implement JSON Decimal support, we need to update Connect
+    final BigDecimal decimal;
+    if (DecimalUtil.isDecimal(context.schema) && context.val instanceof NumericNode) {
```

Comment:
Same call to DecimalUtil.isDecimal(context.schema) in if and else, this can be extracted out.